src/math/libraries/NextStep.sol

```diff
@@ -138,19 +131,26 @@ library NextStep {
             data.deltaUserFutureRewardCollateral,
             data.deltaProtocolFutureRewardCollateral
         );
-        (nextState.startAuction, nextState.merge) = mergingAuction(
-            MergeAuctionData({
-                futureBorrow: data.futureBorrow,
-                futureCollateral: data.futureCollateral,
-                futureRewardBorrow: data.futureRewardBorrow,
-                futureRewardCollateral: data.futureRewardCollateral,
-                deltaFutureBorrow: data.deltaFutureBorrow,
-                deltaFutureCollateral: data.deltaFutureCollateral,
-                auctionStep: data.auctionStep,
-                deltaFuturePaymentBorrow: data.deltaFuturePaymentBorrow,
-                deltaFuturePaymentCollateral: data.deltaFuturePaymentCollateral,
-                blockNumber: data.blockNumber
-            })
-        );
+
+        if (data.cases.cmcb + data.cases.cmbc == 1) {
+            nextState.startAuction = mergingAuction(
+                MergeAuctionData({
+                    futureBorrow: data.futureBorrow,
+                    futureCollateral: data.futureCollateral,
+                    futureRewardBorrow: data.futureRewardBorrow,
+                    futureRewardCollateral: data.futureRewardCollateral,
+                    deltaFutureBorrow: data.deltaFutureBorrow,
+                    deltaFutureCollateral: data.deltaFutureCollateral,
+                    auctionStep: data.auctionStep,
+                    deltaFuturePaymentBorrow: data.deltaFuturePaymentBorrow,
+                    deltaFuturePaymentCollateral: data.deltaFuturePaymentCollateral,
+                    blockNumber: data.blockNumber
+                })
+            );
+        } else if (data.cases.ceccb + data.cases.cecbc == 1) {
+            nextState.startAuction = data.blockNumber;
+        } else {
+            nextState.startAuction = data.blockNumber - data.auctionStep;
```

Comment:
I think here should be 

```
nextState.startAuction = startAuction from state
```

I mean change nothing

Anyway it's impossible to have no caseses we have just

```
nextState.startAuction = startAuction from state
```